cmd/katapult/networks_test.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net"
+	"net/http"
+	"testing"
+
+	"github.com/krystal/go-katapult/core"
+	"github.com/krystal/katapult-cli/config"
+	"github.com/stretchr/testify/assert"
+)
+
+var idNetworks = []*core.Network{
+	{
+		ID:        "pognet",
+		Name:      "Pognet 1",
+		Permalink: "pog-1",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+	{
+		ID:        "pognet2",
+		Name:      "Pognet 2",
+		Permalink: "pog-2",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+}
+
+var subdomainNetworks = []*core.Network{
+	{
+		ID:        "pognet3",
+		Name:      "Pognet 3",
+		Permalink: "pog-3",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+	{
+		ID:        "pognet4",
+		Name:      "Pognet 4",
+		Permalink: "pog-4",
+		DataCenter: &core.DataCenter{
+			ID:        "POG1",
+			Name:      "Pogland 1",
+			Permalink: "pog1",
+			Country: &core.Country{
+				ID:   "pog",
+				Name: "Pogland",
+			},
+		},
+	},
+}
+
+func mockNetworksServer() (net.Listener, error) {
+	ln, err := net.Listen("tcp", "127.0.0.1:7654")
+	if err != nil {
+		return nil, err
+	}
+	mux := &http.ServeMux{}
+	notFound := func(writer http.ResponseWriter, request *http.Request) {
+		writer.WriteHeader(404)
+		_, _ = writer.Write([]byte("Not found."))
+	}
+	mux.HandleFunc("/", notFound)
+	mux.HandleFunc("/core/v1/organizations/_/available_networks", func(writer http.ResponseWriter, request *http.Request) {
+		if request.URL.Path != "/core/v1/organizations/_/available_networks" {
+			notFound(writer, request)
+			return
+		}
+		var networks []*core.Network
+		if id := request.URL.Query().Get("organization[id]"); id == "pog-id" {
+			networks = idNetworks
+		} else if id = request.URL.Query().Get("organization[sub_domain]"); id == "pog-subdomain" {
+			networks = subdomainNetworks
+		} else {
+			notFound(writer, request)
+			return
+		}
+		b, err := json.Marshal(map[string]interface{}{"networks": networks})
+		if err != nil {
+			panic(err)
+		}
+		writer.Header().Add("Content-Type", "application/json")
+		writer.WriteHeader(200)
+		_, _ = writer.Write(b)
+	})
+	go func() { _ = http.Serve(ln, mux) }()
+	return ln, nil
+}
+
+var expectedNetworkResults = []struct {
```

Comment:
Nice test table. Usually I'd expect to see it inside of the test function itself, and usually named "tests".